File: typeface-views/src/main/java/com/sorcererxw/typefaceviews/preferences/ColorStateLists.java
package com.sorcererxw.typefaceviews.preferences;

import android.content.Context;
import android.content.res.ColorStateList;
import android.content.res.Resources;
import android.graphics.Color;
import android.support.v4.content.ContextCompat;
import android.support.v4.graphics.ColorUtils;
import android.util.TypedValue;

/**
 * @description:
 * @author: Sorcerer
 * @date: 2016/12/17
 */

class ColorStateLists {
    static ColorStateList TitleColorStateList(Context context) {
        TypedValue primaryText = new TypedValue();
        Resources.Theme theme = context.getTheme();
        theme.resolveAttribute(android.R.attr.textColorPrimary, primaryText, true);


        return new ColorStateList(
                new int[][]{
                        new int[]{android.R.attr.state_enabled},
                        new int[]{-android.R.attr.state_enabled},
                },
                new int[]{
                        ContextCompat.getColor(context, primaryText.resourceId),
                        ColorUtils.setAlphaComponent(
                                ContextCompat.getColor(context, primaryText.resourceId), 64)
                }
        );
    }

    static ColorStateList summaryColorStateList(Context context) {
        TypedValue primaryText = new TypedValue();
        Resources.Theme theme = context.getTheme();
        theme.resolveAttribute(android.R.attr.textColorSecondary, primaryText, true);

        return new ColorStateList(
                new int[][]{
                        new int[]{android.R.attr.state_enabled},
                        new int[]{-android.R.attr.state_enabled},
                },
                new int[]{
                        ContextCompat.getColor(context, primaryText.resourceId),
                        ColorUtils.setAlphaComponent(
                                ContextCompat.getColor(context, primaryText.resourceId), 64)
                }
        );
    }
}

File: typeface-views/src/main/java/com/sorcererxw/typefaceviews/preferences/ColorStateLists.kt
package com.sorcererxw.typefaceviews.preferences

import android.content.Context
import android.content.res.ColorStateList
import android.content.res.Resources
import android.graphics.Color
import androidx.core.content.ContextCompat
import androidx.core.graphics.ColorUtils
import android.util.TypedValue

/**
 * @description:
 * @author: Sorcerer
 * @date: 2016/12/17
 */

internal object ColorStateLists {
    fun TitleColorStateList(context: Context): ColorStateList {
        val primaryText = TypedValue()
        val theme = context.theme
        theme.resolveAttribute(android.R.attr.textColorPrimary, primaryText, true)


        return ColorStateList(
                arrayOf(intArrayOf(android.R.attr.state_enabled),
                        intArrayOf(-android.R.attr.state_enabled)),
                intArrayOf(ContextCompat.getColor(context, primaryText.resourceId),
                        ColorUtils.setAlphaComponent(
                                ContextCompat.getColor(context, primaryText.resourceId), 64))
        )
    }

    fun summaryColorStateList(context: Context): ColorStateList {
        val primaryText = TypedValue()
        val theme = context.theme
        theme.resolveAttribute(android.R.attr.textColorSecondary, primaryText, true)

        return ColorStateList(
                arrayOf(intArrayOf(android.R.attr.state_enabled),
                        intArrayOf(-android.R.attr.state_enabled)),
                intArrayOf(ContextCompat.getColor(context, primaryText.resourceId),
                        ColorUtils.setAlphaComponent(
                                ContextCompat.getColor(context, primaryText.resourceId), 64))
        )
    }
}
